fmt/src/formatter.rs

```diff
@@ -1148,6 +1148,9 @@ impl<'a, W: Write> Visitor for Formatter<'a, W> {
             None => false,
         })?;
 
+        // EOF newline
+        writeln!(self.buf())?;
+
```

Comment:
put it at the end of `visit_source_unit` after the comments

Reply:
Moved. Thanks!